chore/code quality #1

MildlyBrain opened the pull request:
### Unfinished code refactoring, but should function enough to allow integration with frontend.
### Will continue to work on this branch, until complete refactoring of the entire backend.
## Features
- Queue route and controller
- Queue model update to have status
- Removed Restaurants virtuals
- Ensure user populate on reservation, queue, and reviews
- Setup helper functions and added error handling middleware

MildlyBrain committed:
feat: helper functions and class

MildlyBrain committed:
feat: errorHandler middleware and pagination metadata

MildlyBrain committed:
style: error message and error response

MildlyBrain committed:
refactor: restaurant controller

MildlyBrain committed:
feat: remove restaurant virtuals

MildlyBrain committed:
refactor: part of reservations controller

MildlyBrain committed:
feat: queue controller with status in model

MildlyBrain committed:
feat: queue routes and queue delete role check

MildlyBrain committed:
feat, style: prettier'd everything and add user populate

MildlyBrain committed:
feat: getQueuePosition now possible

MildlyBrain committed:
fix: remove populate on restaurants

MildlyBrain committed:
fix: update wrong field and not working delete

MildlyBrain committed:
fix: skill issue

MildlyBrain committed:
feat: handle queueStatus not given

ShaneChaneShane commented on controllers/queues.js:
```diff
@@ -0,0 +1,130 @@
+  if (
+    (req.user.role === "user" && queue.user.toString() !== req.user.id) ||
+    (req.user.role === "employee" &&
+      queue.restaurant.toString() !== req.user.employedAt)
```
```queue.restaurant.toString() !== req.user.employedAt.toString()```

ShaneChaneShane reviewed (changes requested):
unimplemented
- create queue: user should be able to create only 1 queue per a restaurant
- create queue: queue's seatCount should consider restaurants' seatPerReservationLimit
- getQueuePosition: queue position should consider ahead queues' status (if some of the queues ahead are 'completed', the position should move up)

MildlyBrain committed:
fix: queue condition and missing populate

- Queue delete condition was not comparing properly
- GET Reviews missing populate user
- Menu has unused enum
- Allow only user to POST queue

ShaneChaneShane reviewed (changes requested):
pls populate getQueues with restaurant too the same way to getReservations pls yay
modify here
https://github.com/2110503-2564/se-project-backend-24_scam/blob/da822441dc7a30e46377ca704160f9ba7f2c86bc/controllers/queues.js#L19-L22

like getReservations
https://github.com/2110503-2564/se-project-backend-24_scam/blob/da822441dc7a30e46377ca704160f9ba7f2c86bc/controllers/reservations.js#L16-L19 https://github.com/2110503-2564/se-project-backend-24_scam/blob/da822441dc7a30e46377ca704160f9ba7f2c86bc/controllers/reservations.js#L46-L49

MildlyBrain committed:
style: add prettier config

MildlyBrain committed:
fix: minor queue fixes per suggestion

MildlyBrain committed:
refactor,style: getReservation refactor and ran prettier again

ShaneChaneShane committed:
fix: employee delete queue minor logic fix

ShaneChaneShane reviewed (approved):
look good now bro
![200w (1).gif](https://github.com/user-attachments/assets/da226acb-48de-45c5-8a61-2d0bb43c4501)
